packages/shared/src/libs/types/permissions.type.ts

```diff
@@ -0,0 +1,9 @@
+import { type PermissionKey } from "../enums/enums.js";
+import { type ValueOf } from "./types.js";
+
+type Permissions = {
```

Comment:
Maybe instead we can pass permissions in the following format: 
```
[PermissionKey, [PermissionKey,PermissionKey]]
```

Reply:
allow either permission key or array, if it's array then one of the permission from array is required

Reply:
but this was the type I initially add and you said it's very confusing... 🙈😅
https://github.com/BinaryStudioAcademy/bsa-2024-gitfit/pull/414#discussion_r1766868924

Reply:
Maybe I didn't get the point you meant in that comment?

Reply:
Or maybe I didn't get your current suggession..?

Initially I made it work in such a way, that you should pass an array containing elements of type `PermissinKey | PermissionKey[]` so user must have ALL of this passed permissions. And thanks to `PermissionKey[]` we can list permissions that user must have AT LEAST one of them. So for example if we pass such an array: [A, [B,C]] the function will return true if user have two permissions: A & at least one of [B, C].

Haven't you suggest the same right now?

Reply:
I meant to pass it as array instead of an object. But now I'm think maybe we don't need this at all. Why can't we just call checkPermissions twice? like checkHasPermissions(requiredPermission) &  checkHasPermissions([permission1, permission2]). And for the backend hooks it will be just [checkHasPermissions(requiredPermission), checkHasPermissions([permission1, permission2])]

Reply:
Yeap. I also think so. Then I'll revert all the changes and implement this easier solution

Reply:
Thanks